### ISSUE TITLE
Nightly issue, mutex?

### ISSUE THREAD

**Issue by jasonrichardsmith** [nanovms/nanos#1661]
I have to use nightly right now because I need this code.
https://github.com/nanovms/nanos/commit/c3a488e4be78e8697fc422f05f6cb934d2192bf9

VMS I provisioned the other day are fine but they are now failing on nightly. I am getting errors below.

```
ops pkg load python_3.8.6 -c config.json -n
 100% |████████████████████████████████████████|  [0s:0s]
warning: overwriting existing file /etc/resolv.conf hostpath old: /tmp/python3_temp328684329/resolv.conf new: etc/resolv.conf
booting /home/jason/.ops/images/python3 ...
en1: assigned 10.0.2.15
/.local/lib/python3.8/site-packages/googlecloudprofiler/client.py:167: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(self._profilers) is 0:
{"timestamp": 1643296662.4657683, "level": null, "name": "emailservice-server", "message": "starting the email service in dummy mode.", "severity": "INFO"}
{"timestamp": 1643296662.4674828, "level": null, "name": "emailservice-server", "message": "Profiler disabled.", "severity": "INFO"}
{"timestamp": 1643296662.4689507, "level": null, "name": "emailservice-server", "message": "Tracing enabled.", "severity": "INFO"}
en1: assigned FE80::BC72:AEFF:FEC3:7D6
{"timestamp": 1643296671.4780874, "level": null, "name": "emailservice-server", "message": "Tracing disabled.", "severity": "INFO"}
E0127 15:17:51.493388929       1 cpu_linux.cc:50]            Cannot determine number of CPUs: assuming 1
epoll_ctl error: add: EPOLLEXCLUSIVE not supported
epoll_ctl error: add: EPOLLEXCLUSIVE not supported
D0127 15:17:51.496470846       1 is_epollexclusive_available.cc:100] epoll_ctl with EPOLLEXCLUSIVE failed with error: 22. Not using epollex polling engine.
I0127 15:17:51.498017771       1 ev_epollex_linux.cc:1624]   Skipping epollex because it is not supported.
I0127 15:17:51.499028795       1 ev_epoll1_linux.cc:122]     grpc epoll fd: 3
D0127 15:17:51.499764070       1 ev_posix.cc:173]            Using polling engine: epoll1
D0127 15:17:51.502260830       1 lb_policy_registry.cc:42]   registering LB policy factory for "grpclb"
D0127 15:17:51.503279326       1 lb_policy_registry.cc:42]   registering LB policy factory for "priority_experimental"
D0127 15:17:51.504649637       1 lb_policy_registry.cc:42]   registering LB policy factory for "weighted_target_experimental"
D0127 15:17:51.505882838       1 lb_policy_registry.cc:42]   registering LB policy factory for "pick_first"
D0127 15:17:51.506896055       1 lb_policy_registry.cc:42]   registering LB policy factory for "round_robin"
D0127 15:17:51.507917191       1 dns_resolver_ares.cc:507]   Using ares dns resolver
D0127 15:17:51.509350850       1 lb_policy_registry.cc:42]   registering LB policy factory for "cds_experimental"
D0127 15:17:51.510519389       1 lb_policy_registry.cc:42]   registering LB policy factory for "eds_experimental"
D0127 15:17:51.511606446       1 lb_policy_registry.cc:42]   registering LB policy factory for "eds_drop_experimental"
D0127 15:17:51.512851158       1 lb_policy_registry.cc:42]   registering LB policy factory for "xds_cluster_manager_experimental"
{"timestamp": 1643296671.5159767, "level": null, "name": "emailservice-server", "message": "listening on port: 5055", "severity": "INFO"}
ffffc0000cbdff60:   ffffffff8004343b	(mutex_lock + 000000000000013b/000000000000015a)
ffffc0000cbdff70:   ffffffff800558dc	(socket + 000000000000018c/000000000000025f)
ffffc0000cbdffb0:   ffffffff800c5175	(syscall_handler + 0000000000000165/000000000000058a)
assertion !ctx->next_waiter failed at /home/circleci/project/src/kernel/mutex.c:58  in mutex_lock_internal(); halt
```
Then crash

**Comment by wjhun** [comment]
> I have to use nightly right now because I need this code.
> c3a488e <https://github.com/nanovms/nanos/commit/c3a488e4be78e8697fc422f05f6cb934d2192bf9>
> VMS I provisioned the other day are fine but they are now failing on nightly. I am getting errors below.
> 
Jason,

Thanks for the heads up and apologies for the inconvenience. We've reproduced the problem and are working on a fix at the moment.

We're also working on adding the --nanos-version flag to 'ops pkg load' so that you can build an image using build artifacts from a specific commit. This will allow you to revert to using a known working commit when issues like these come up in the nightly build.

Best,
Will

**Comment by wjhun** [comment]
This issue should be resolved with https://github.com/nanovms/nanos/commit/eeb8acc73944cb50546155c9d06bb260c01586dc and picked up by tonight's nightly build.

If you'd like to run it sooner, you could build ops from the master tree and specify a nanos commit ID when running 'ops pkg load':

```
$ ops pkg load --nanos-version eeb8acc python_3.8.6 -c config.json
```

**Comment by wjhun** [comment]
Resolved in #1662
